dali/kernels/tensor_view.h

```diff
@@ -69,9 +69,9 @@ bool ContainsCoords(const Shape &shape, const Position &pos) {
 /// @brief Calculates flat index of a given element in the tensor
 /// @remarks If pos has fewer dimensions than shape, the remaining offsets are assumed to be 0
 template <typename Shape, typename Position>
-ptrdiff_t CalcOffset(const Shape &shape, const Position &pos) {
+if_array_like<Position, ptrdiff_t> CalcOffset(const Shape &shape, const Position &pos) {
   ptrdiff_t ofs = pos[0];
-  const int pos_dim = ShapeDim(pos);
+  const int pos_dim = size(pos);
   const int shape_dim = ShapeDim(shape);
```

Comment:
If we already introduced `size`, you could remove `ShapeDim` and swap it for `size`.

Reply:
`size` is undefined for scalar values, `ShapeDim` gives 1.